### DISCUSSION TITLE
curl_easy_perform() failed:Failure when receiving data from the peer:main:93

### DISCUSSION THREAD

**Discussion by samnaction** [curl/curl#8277] [General]
When you build libcurl in Visual Studio solution from `.\projects\Windows` be sure to remove `DEBUGBUILD` preprocessor to avoid the above error.

The issue happens when you create a simple HTTPS server using a self-signed certificate.

Steps:

1. Build debug libcurl from `.\projects\Windows` with open SSL enabled
2. Create a simple HTTPS Server using a self-signed certificate
3. Make an HTTPS call using libcurl Eg: https://curl.se/libcurl/c/https.html
4. Error

Confusion: 
Here are a few things which took more time for me to fix this issue.

1. When you build libcurl from `.\winbuild` using `nmake /f Makefile.vc` the error won't happen. This made me think there is something wrong with the visual studio solution and I was looking into other preprocessors. I never doubted `DEBUGBUILD`. This is happening because `nmake` is not adding this preprocessor even though if you say `DEBUG=yes`

2. When you make a call to https://www.google.com, the code will work fine. This made me look into the self-signed certificates and my HTTPS Server. Apparently, the issue may be with HTTPS Server sending SSL close_notify.

3. Not using curl error buff. I was relying on `curl_easy_strerror(res)` 
Always use this 

`char errbuf[CURL_ERROR_SIZE];
curl_easy_setopt(eh, CURLOPT_ERRORBUFFER, errbuf);`

The third point gave me the clue to identify the issue.

Hope this helps someone. So posting here.

**Comment by jay** [comment]
What is the error message?
